### PR TITLE
Magento 2.4.* fix

### PR DESCRIPTION
**Pull request by kylianleemkuil**
+ Renamed some keys for Magento 2.4.*
+ Deleted some Constraints for Magento 2.4.*
+ Added checks for doubles in tables

### CODE REVIEW

**Comment by thomas-kl1** [comment]
I'm going to review this later tonight! Thanks for contributing!! :)

**Review by thomas-kl1 — CHANGES_REQUESTED** [review]
Overall looks fine to me :)
Could you check why you removed the drop on row_id column in some cases?

File: scripts/product.sql [hunk 1]
```diff
@@ -287,81 +287,84 @@ ALTER TABLE `catalog_product_bundle_selection`
...
+    DROP INDEX `CATALOG_PRODUCT_ENTITY_VARCHAR_ROW_ID_ATTRIBUTE_ID_STORE_ID`,
     ADD CONSTRAINT `CATALOG_PRODUCT_ENTITY_VARCHAR_ENTITY_ID_ATTRIBUTE_ID_STORE_ID` UNIQUE KEY (`entity_id`,`attribute_id`,`store_id`),
     DROP COLUMN `row_id`;
 
 -- Gallery value to entity
 ALTER TABLE `catalog_product_entity_media_gallery_value_to_entity`
     DROP FOREIGN KEY `CAT_PRD_ENTT_MDA_GLR_VAL_TO_ENTT_ROW_ID_CAT_PRD_ENTT_ROW_ID`,
-    DROP INDEX `CAT_PRD_ENTT_MDA_GLR_VAL_TO_ENTT_VAL_ID_ENTT_ID`,
-    ADD CONSTRAINT `CAT_PRD_ENTT_MDA_GLR_VAL_TO_ENTT_VAL_ID_ENTT_ID` UNIQUE KEY (`value_id`,`entity_id`),
-    DROP COLUMN `row_id`;
```
[the remark this answers is not in the document]
Comment (kylianleemkuil) [reply scripts/product.sql:337]: My bad deleted that for my own test reasons.

File: scripts/product.sql [hunk 2]
```diff
@@ -287,81 +287,84 @@ ALTER TABLE `catalog_product_bundle_selection`
...
 
 -- Gallery value
 ALTER TABLE `catalog_product_entity_media_gallery_value`
     DROP FOREIGN KEY `CAT_PRD_ENTT_MDA_GLR_VAL_ROW_ID_CAT_PRD_ENTT_ROW_ID`,
-    DROP INDEX `CATALOG_PRODUCT_ENTITY_MEDIA_GALLERY_VALUE_ENTITY_ID`,
+    DROP INDEX `CATALOG_PRODUCT_ENTITY_MEDIA_GALLERY_VALUE_ROW_ID`,
+	DROP INDEX `CAT_PRD_ENTT_MDA_GLR_VAL_ENTT_ID_VAL_ID_STORE_ID`,
     ADD INDEX `CATALOG_PRODUCT_ENTITY_MEDIA_GALLERY_VALUE_ENTITY_ID` (`entity_id`),
-    ADD CONSTRAINT `CAT_PRD_ENTT_MDA_GLR_VAL_ENTT_ID_VAL_ID_STORE_ID` UNIQUE KEY (`entity_id`,`value_id`,`store_id`),
-    DROP COLUMN `row_id`;
```
[the remark this answers is not in the document]
Comment (kylianleemkuil) [reply scripts/product.sql:345]: My bad deleted that for my own test reasons.

File: scripts/product.sql [hunk 3]
```diff
@@ -287,81 +287,84 @@ ALTER TABLE `catalog_product_bundle_selection`
...
 -- Gallery
 ALTER TABLE `catalog_product_entity_gallery`
     DROP FOREIGN KEY `CAT_PRD_ENTT_GLR_ROW_ID_CAT_PRD_ENTT_ROW_ID`,
-    DROP INDEX `CATALOG_PRODUCT_ENTITY_GALLERY_ENTITY_ID`,
-    DROP INDEX `CATALOG_PRODUCT_ENTITY_GALLERY_ENTITY_ID_ATTRIBUTE_ID_STORE_ID`,
+    DROP INDEX `CATALOG_PRODUCT_ENTITY_GALLERY_ROW_ID`,
+    DROP INDEX `CATALOG_PRODUCT_ENTITY_GALLERY_ROW_ID_ATTRIBUTE_ID_STORE_ID`,
     ADD INDEX `CATALOG_PRODUCT_ENTITY_GALLERY_ENTITY_ID` (`entity_id`),
-    ADD CONSTRAINT `CATALOG_PRODUCT_ENTITY_GALLERY_ENTITY_ID_ATTRIBUTE_ID_STORE_ID` UNIQUE KEY (`entity_id`,`attribute_id`,`store_id`),
-    DROP COLUMN `row_id`;
```
[the remark this answers is not in the document]
Comment (kylianleemkuil) [reply scripts/product.sql:354]: My bad deleted that for my own test reasons.

File: scripts/product.sql [hunk 4]
```diff
@@ -287,81 +287,84 @@ ALTER TABLE `catalog_product_bundle_selection`
...
+    ADD CONSTRAINT `CATALOG_PRODUCT_ENTITY_GALLERY_ENTITY_ID_ATTRIBUTE_ID_STORE_ID` UNIQUE KEY (`entity_id`,`attribute_id`,`store_id`);
 
 -- Tier price
 ALTER TABLE `catalog_product_entity_tier_price`
     DROP FOREIGN KEY `CAT_PRD_ENTT_TIER_PRICE_ROW_ID_CAT_PRD_ENTT_ROW_ID`,
-    DROP INDEX `UNQ_E8AB433B9ACB00343ABB312AD2FAB087`,
-    ADD CONSTRAINT `UNQ_E8AB433B9ACB00343ABB312AD2FAB087` UNIQUE KEY (`entity_id`,`all_groups`,`customer_group_id`,`qty`,`website_id`),
-    DROP COLUMN `row_id`;
```
[the remark this answers is not in the document]
Comment (kylianleemkuil) [reply scripts/product.sql:361]: My bad deleted that for my own test reasons.

**Comment by kylianleemkuil** [comment]
I will make a new pull request

**Comment by thomas-kl1** [comment]
@kylianleemkuil every new commits you push to your branch used in this PR will be taken. So you don't need to open a new PR

**Pull request closed without merging**